upgrade.js

```diff
@@ -90,27 +90,23 @@ function main (knownUpgrades) {
 
   var elm = localFindBinary(
     'elm',
-    'ERROR: elm was not found on your PATH.  Make sure you have Elm 0.18 installed.\n' + howToInstallElm()
+    'ERROR: elm was not found on your PATH.  Make sure you have Elm 0.19 installed.\n' + howToInstallElm()
   )
 
-  var elmUsage = childProcess.execFileSync(elm)
-  var elmVersion = elmUsage.toString().split('\n')[0].split(' - ')[0].trim()
-  if (!elmVersion.match(/^Elm Platform 0\.18\./)) {
-    process.stderr.write('ERROR: Elm 0.18 required, but found ' + elmVersion + '\n' + howToInstallElm())
+  var elmUsage = childProcess.execFileSync(elm, [ '--version' ])
+  var elmVersion = elmUsage.toString().split('\n')[0]
+  if (!elmVersion.match(/^0\.19\./)) {
+    process.stderr.write('ERROR: Elm 0.19 required, but found ' + elmVersion + '\n' + howToInstallElm())
     process.exit(1)
   }
-  var elmPackageBinary = localFindBinary(
-    'elm-package',
-    'ERROR: elm-package was not found on your PATH.  Make sure you have Elm 0.18 installed.\n' + howToInstallElm()
-  )
-  process.stdout.write('INFO: Found ' + elmVersion + '\n')
+  process.stdout.write('INFO: Found elm ' + elmVersion + '\n')
 
   var elmFormat = localFindBinary('elm-format', 'ERROR: elm-format was not found on your PATH.  Make sure you have elm-format installed.\n' + howToInstallElmFormat())
 
   var elmFormatUsage = childProcess.execFileSync(elmFormat)
   var elmFormatVersion = elmFormatUsage.toString().split('\n')[0].trim().split(' ')[1]
-  if (semver.lt(elmFormatVersion, '0.5.2-alpha')) {
-    process.stderr.write('ERROR: elm-format >= 0.5.2-alpha required, but found ' + elmFormatVersion + '\n' + howToInstallElmFormat())
+  if (semver.lt(elmFormatVersion, '0.7.1-beta')) {
```

Comment:
:thought_balloon: In theory we can abstract this out. If we had a build script that injected the version, then publishing a new version of elm-upgrade would be trivial

Reply:
I think that will be worth thinking about for 0.20.  At the moment, it hasn't seemed like there's a ton of consistency between the 0.18 and 0.19 upgrades, so I suspect it's likely future Elm versions upgrades will still need a lot of manual attention to upgrade elm-upgrade.